test/kubernetes_unittest.cc

```diff
@@ -1215,4 +1184,705 @@ TEST_F(KubernetesTestFakeServer, MetadataQuery) {
+    EXPECT_FALSE(k8s_pod_metadata.ignore);
+    EXPECT_EQ("TestVersion", k8s_pod_metadata.version);
+    EXPECT_FALSE(k8s_pod_metadata.is_deleted);
+    EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+              k8s_pod_metadata.created_at);
+    EXPECT_EQ(PodMetadataWithInstance("TestPodName2", pod2)->ToString(),
+              k8s_pod_metadata.metadata->ToString());
+
+    const auto& k8s_container_metadata = metadata_map.at(k8s_container_resource2);
+    EXPECT_FALSE(k8s_container_metadata.ignore);
+    EXPECT_EQ("1.6", k8s_container_metadata.version);
+    EXPECT_FALSE(k8s_container_metadata.is_deleted);
+    EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+              k8s_container_metadata.created_at);
+    EXPECT_EQ(ContainerMetadataWithInstance("TestPodName2")->ToString(),
+              k8s_container_metadata.metadata->ToString());
+
+    last_timestamp = k8s_pod_metadata.collected_at;
+  }
+
+  // Delete pod #1 and wait until watch has processed it (by polling the store).
+  server.SendStreamResponse(pods_watch_path, json::object({
+    {"type", json::string("DELETED")},
+    {"object", pod1->Clone()},
+  })->ToString());
+  EXPECT_TRUE(WaitForNewerCollectionTimestamp(
+      store, k8s_pod_resource1, last_timestamp));
+
+  // Verify pod #1 values in the store.
+  EXPECT_EQ(k8s_pod_resource1, store.LookupResource(
+      "k8s_pod.TestNamespace.TestPodName1"));
+  EXPECT_EQ(k8s_pod_resource1, store.LookupResource(
+      "k8s_pod.TestPodUid1"));
+  EXPECT_EQ(k8s_container_resource1, store.LookupResource(
+      "k8s_container.TestNamespace.TestPodName1.TestContainerName0"));
+  EXPECT_EQ(k8s_container_resource1, store.LookupResource(
+      "k8s_container.TestPodUid1.TestContainerName0"));
+  {
+    const auto metadata_map = store.GetMetadataMap();
+
+    const auto& k8s_pod_metadata = metadata_map.at(k8s_pod_resource1);
+    EXPECT_FALSE(k8s_pod_metadata.ignore);
+    EXPECT_EQ("TestVersion", k8s_pod_metadata.version);
+    EXPECT_TRUE(k8s_pod_metadata.is_deleted);
+    EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+              k8s_pod_metadata.created_at);
+    EXPECT_EQ(PodMetadataWithInstance("TestPodName1", pod1)->ToString(),
+              k8s_pod_metadata.metadata->ToString());
+
+    const auto& k8s_container_metadata = metadata_map.at(k8s_container_resource1);
+    EXPECT_FALSE(k8s_container_metadata.ignore);
+    EXPECT_EQ("1.6", k8s_container_metadata.version);
+    EXPECT_TRUE(k8s_container_metadata.is_deleted);
+    EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+              k8s_container_metadata.created_at);
+    EXPECT_EQ(ContainerMetadataWithInstance("TestPodName1")->ToString(),
+              k8s_container_metadata.metadata->ToString());
+  }
+}
+
+// Wait for updater's service & endpoints watchers to connect to the
+// server (hanging GET), then send 6 stream responses from the fake
```

Comment:
Do you want to add a breakdown of responses, the same as in `TestNodes` and `TestPods`?

Reply:
Done.